src/braket/experimental/autoqasm/operators/conditional_expressions.py

```diff
@@ -47,15 +48,28 @@ def _oqpy_if_exp(
     if_true: Callable[[None], Any],
     if_false: Callable[[None], Any],
     expr_repr: Optional[str],
-) -> None:
+) -> Optional[oqpy.base.Var]:
     """Overload of if_exp that stages an oqpy conditional."""
-    oqpy_program = program.get_program_conversion_context().get_oqpy_program()
+    oqpy_program = aq_program.get_program_conversion_context().get_oqpy_program()
     if isinstance(cond, oqpy.base.Var) and cond.name not in oqpy_program.declared_vars.keys():
         oqpy_program.declare(cond)
+
+    result_var = None
     with oqpy.If(oqpy_program, cond):
-        if_true()
```

Comment:
Question for my education: Why does this solve the issue? And why we don't need to do the same in if-else in control_flow.py

Reply:
For conditional expressions, the entire expression (`aq.IntVar(1) if foo else aq.IntVar(2)`) is supposed to be resolved to a single value. However, OpenQASM does not support inline conditional expressions. So we have to convert this into a traditional `if`/`else` block in OpenQASM. But because the result of the conditional expression may be assigned to something, we need to do an assignment in both branches as well.

For the `if`/`else` in control_flow.py, the user has already written it as an if-else block with the appropriate assignment statements in each block. So we don't have to do this type of translation there.